### PR TITLE
Rails6.0 upgrade

### PR DESCRIPTION
**Pull request by legendarylarry**
rails 6.0.0

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -56,7 +56,7 @@ group :development, :test do
 end
 
 #coverage map for testing
-gem 'simplecov', :require => false, :group => :test
+# gem 'simplecov', :require => false, :group => :test
```
Comment (cgokey) [inline Gemfile:59]: This should not be commented out

File: Gemfile [hunk 2]
```diff
@@ -11,15 +11,15 @@ gem 'webpacker'
...
-gem 'bourbon'
-gem 'neat'
+gem 'bourbon', '4.2.6'
+gem 'neat', '1.7.2'
 
 # Use Uglifier as compressor for JavaScript assets
 gem 'uglifier', '>= 1.3.0'
 # Use CoffeeScript for .coffee assets and views
-gem 'coffee-rails', '~> 4.2.2'
+gem 'coffee-rails' # , '~> 4.2.2'
```
Comment (cgokey) [inline Gemfile:22]: Remove this comment or use a specific version.

File: Gemfile [hunk 3]
```diff
@@ -89,7 +90,7 @@ gem 'rails-controller-testing'
 group :development do
   # Access an IRB console on exception pages or by using <%= console %> in views
   gem 'web-console'
-
+  gem 'listen'
```
Comment (cgokey) [inline Gemfile:93]: is this necessary?

File: app/javascript/components/model/PaginationTableModel.js [hunk 4]
```diff
@@ -2,17 +2,20 @@ import { computed, observable, decorate} from "mobx"
...
-  colorCode = null // String
-  sortColumn = null // String
-  sortOrder = "asc" // String
-  currentPage = 1 // Int
-  result = null // Result
-  loading = false // Bool
-  pageSize = 10 // Int
+
+  constructor() {
+    this.section = null // String
```
Comment (cgokey) [inline app/javascript/components/model/PaginationTableModel.js:7]: Did the other way fail?

File: app/javascript/components/view/PaginationTableView.js [hunk 5]
```diff
@@ -7,14 +7,21 @@ import PropTypes from "prop-types";
 import {observer} from "mobx-react"
 import "../eui.css"
 import "../pagination.css"
-import ColorFilterView from "./ColorFilterView";
+// import ColorFilterView from "./ColorFilterView";
```
Comment (cgokey) [inline app/javascript/components/view/PaginationTableView.js:10]: I think this should be put back to the way it was.

File: app/views/shared/_records_table.html.erb [hunk 6]
```diff
@@ -25,6 +25,7 @@
 
     <div class="results-table-container">
       <div class="table-scroll">
+        My React Component
```
Comment (cgokey) [inline app/views/shared/_records_table.html.erb:28]: Remove this, test string left in from me.

File: config/application.rb [hunk 7]
```diff
@@ -1,7 +1,6 @@
 require_relative 'boot'
 
 require 'rails/all'
-require  './app/middleware/middleware_healthcheck'
```
Comment (cgokey) [inline config/application.rb:4]: I think we'll want to keep the middleware health check

File: config/application.rb [hunk 8]
```diff
@@ -10,44 +9,32 @@
...
-    # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
-    # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
-    # config.i18n.default_locale = :de
-
-    # config.session_store :cookie_store, key: '_interslice_session'
-    # config.middleware.use ActionDispatch::Cookies # Required for all session management
-    # config.middleware.use ActionDispatch::Session::CookieStore, config.session_options
-
-    config.middleware.insert_after Rails::Rack::Logger, MiddlewareHealthcheck
+    # config.middleware.insert_after Rails::Rack::Logger, MiddlewareHealthcheck
```
Comment (cgokey) [inline config/application.rb:21]: Likewise, middleware health check should not be removed.

File: config/environments/development.rb [hunk 9]
```diff
@@ -53,25 +52,16 @@
...
 
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
-  # config.file_watcher = ActiveSupport::EventedFileUpdateChecker
-  # cmr base url
-  config.cmr_base_url = 'https://cmr.sit.earthdata.nasa.gov'
+  config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 
-  config.kms_base_url = 'https://gcmd.sit.earthdata.nasa.gov'
+  config.cmr_base_url = 'https://cmr.earthdata.nasa.gov'
```
Comment (cgokey) [inline config/environments/development.rb:63]: These should be put back to using sit

File: Gemfile (hunk 1 above)
Comment (legendarylarry) [reply Gemfile]: fixed

File: Gemfile (hunk 2 above)
Comment (legendarylarry) [reply Gemfile]: fixed

File: Gemfile (hunk 3 above)
Comment (legendarylarry) [reply Gemfile]: removed

File: app/javascript/components/model/PaginationTableModel.js (hunk 4 above)
Comment (legendarylarry) [reply app/javascript/components/model/PaginationTableModel.js:7]: Not sure, when I checkout out your branch and ran, I had no failures but ~40 errors IIRC. I had in  my pocket a branch that one of my devs had been working on the react stuff, and when I merged yours and theirs all was well.

File: app/javascript/components/view/PaginationTableView.js (hunk 5 above)
Comment (legendarylarry) [reply app/javascript/components/view/PaginationTableView.js]: back

File: config/application.rb (hunk 7 above)
Comment (legendarylarry) [reply config/application.rb:5]: fixed

File: app/views/shared/_records_table.html.erb (hunk 6 above)
Comment (legendarylarry) [reply app/views/shared/_records_table.html.erb]: removed

File: app/javascript/components/model/Result.js [hunk 10]
```diff
@@ -1,22 +1,26 @@
...
 
-    let records = json["records"]
-    for (let i=0; i<records.length; i++) {
-      records[i].no = (this.itemsPerPage*(this.currentPage-1))+(i+1)
+    this.totalItems = null // Int
+    this.currentPage = 1 // Int
+    this.itemsPerPage = null // Int
+    this.records = [] // [String:Any]
+
+    if(json) {
```
Comment (cgokey) [inline app/javascript/components/model/Result.js:12]: Why did you add a "if (json)" this is different than the original logic?   should this be removed?

File: app/javascript/components/view/PaginationTableView.js [hunk 11]
```diff
@@ -9,12 +9,19 @@ import "../eui.css"
 import "../pagination.css"
 import ColorFilterView from "./ColorFilterView";
 
+// class PaginationTableView extends React.Component {
```
Comment (cgokey) [inline app/javascript/components/view/PaginationTableView.js:12]: Remove this comment

File: config/environments/development.rb [hunk 12]
```diff
@@ -10,32 +10,31 @@
...
+
   # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
-  config.action_mailer.perform_deliveries = false
 
   config.action_mailer.perform_caching = false
-  config.action_mailer.delivery_method = :file
-  config.action_mailer.file_settings = {
-    location: 'log/emails'
-  }
```
[the remark this answers is not in the document]
Comment (cgokey) [reply config/environments/development.rb:38]: I don't know why the rails update removed these settings, but I think we want to keep this.

File: config/environments/production.rb [hunk 13]
```diff
@@ -18,78 +18,53 @@
...
+  # config.active_job.queue_name_prefix = "cmr_metadata_review_production"
+
   config.action_mailer.perform_caching = false
 
   # Ignore bad email addresses and do not raise email delivery errors.
   # Set this to true and configure the email server for immediate delivery to raise delivery errors.
   # config.action_mailer.raise_delivery_errors = false
-  # config.action_mailer.delivery_method = :smtp
-  # config.action_mailer.perform_deliveries = false
-  config.action_mailer.default_url_options = {host: 'https://cmr-dashboard.earthdata.nasa.gov'}
```
Comment (cgokey) [inline config/environments/production.rb:92]: Wonder if we should add this back too.

File: app/javascript/components/model/Result.js (hunk 10 above)
Comment (legendarylarry) [reply app/javascript/components/model/Result.js:12]: seems like a reasonable safeguard against an impending undefined.length in the for loop at the least. better solution might be to leave it and add a console.warn or similar in the `else` 
But as I'm not a react developer I will just remove it. I'm  not sure why my username is attached to these react changes, must have been from when I recreated the 6.0.0 state for review.

File: test/test_helper.rb [hunk 14]
```diff
@@ -43,11 +29,6 @@
 Capybara.default_driver = :headless_chrome
 Capybara.javascript_driver = :headless_chrome
 
-# Not sure this is the best thing to do, but we don't call localhost for anything needed in our tests.
```
Comment (cgokey) [inline test/test_helper.rb:46]: We don't need this anymore?

File: test/test_helper.rb [hunk 15]
```diff
@@ -10,30 +10,16 @@
...
-Minitest::Reporters.use! [Minitest::Reporters::SpecReporter.new, Minitest::Reporters::JUnitReporter.new]
 
-# Specs flagged with `js: true` will use Capybara's JS driver.
-Capybara.register_driver :headless_chrome do |app|
-  # set timeout to 60s http://www.testrisk.com/2016/05/change-default-timeout-and-wait-time-of.html
-  # need to use read_timeout and open_timeout https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/remote/http/default.rb
-  client = Selenium::WebDriver::Remote::Http::Default.new
-  client.read_timeout = 60
-  client.open_timeout = 60
+# Not sure this is the best thing to do, but we don't call localhost for anything needed in our tests.
```
Comment (cgokey) [inline test/test_helper.rb:14]: oh I see it here.

**Review by cgokey — APPROVED** [review]

**Review by jdf-eos — APPROVED** [review]

**Review by cgokey — APPROVED** [review]
